persistence-api/src/main/java/io/stargate/db/datastore/DataStore.java

```diff
@@ -45,19 +68,11 @@ default QueryBuilder query() {
   CompletableFuture<ResultSet> query(
       String cql, Optional<ConsistencyLevel> consistencyLevel, Object... parameters);
```

Comment:
GraphQL (and other APIs I imagine) needs to set other execution parameters: fetch size, paging state... And it's a bit more convenient to do it at this level, rather than when we create the DataStore.

Instead of just `consistencyLevel`, this method could take a `UnaryOperator<Parameters>` that gives the caller a chance to customize whatever they want.

Reply:
+1 It'd be nice if `DataStore` was a long-lived object and `Parameter` was pass per method instead of a constructor parameter. Maybe this is a follow up issue?

Reply:
Good idea, I'll add that.

Reply:
I'll note that I kept the `Parameters` in the ctor and the `UnaryOperator<Parameters>` approach. I think it's useful to pass your own "long lived default" (and some parameters like the protocol version probably don't ever need to be specified per-query) but be able to modify a few things per-query, which this allows.